libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/arrow/model/RouteArrowUpdateError.kt

```diff
@@ -0,0 +1,33 @@
+package com.mapbox.navigation.ui.maps.route.arrow.model
+
+/**
+ * Represents a [MapboxRouteArrowApi] error condition.
+ *
+ * @param errorMessage an error message describing the error.
+ * @param throwable an optional throwable related to the error.
+ */
+sealed class RouteArrowUpdateError constructor(errorMessage: String, throwable: Throwable?) {
```

Comment:
Are we ok having a `sealed class` here? Should we have separated regular classes with `internal constructor`s instead?

Reply:
It made sense to me because the addArrow method can return specific error types. The child classes have internal constructors. The error type indicates the error better than the text string inside of it IMO.